p2p/src/peers/mod.rs

```diff
@@ -112,6 +112,28 @@ impl Peers {
         contains
     }
 
+    /// Extract all melted peers included in the ice bucket
+    pub fn extract_melted_peers_from_ice_bucket(&mut self) -> Vec<SocketAddr> {
+        let mut addresses = vec![];
+        let new_ice_bucket = self
```

Comment:
You should probably use [Vec::retain](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.retain) instead of clone + filter_map